pilot/pkg/config/coredatamodel/discovery.go

```diff
@@ -79,17 +79,18 @@ func (d *MCPDiscovery) HandleCacheEvents(config model.Config, event model.Event)
 	defer d.cacheMutex.Unlock()
 	services := convertServices(config)
 	d.mergeCachedServices(services)
+	mixerEnabled := d.mixerEnabled()
 	switch event {
 	case model.EventAdd, model.EventUpdate:
-		newSvcInstancesByIP, newSvcInstancesByHost := d.convertInstances(config, services)
+		newSvcInstancesByIP, newSvcInstancesByHost := d.convertInstances(config, services, mixerEnabled)
```

Comment:
why do we need this ? Since mixer will go away soon I would avoid polluting the interfaces.

Reply:
I'm not entirely sure ... this PR just preserves the existing behavior.